scalafmt-core/shared/src/main/scala/org/scalafmt/util/TreeExtractors.scala

```diff
@@ -42,11 +42,8 @@ object `:parent:` {
 object WithChain {
   def unapply(t: Type.With): Option[Type.With] =
     TreeOps.topTypeWith(t) match {
-      // self types
-      case (top: Type.With) `:parent:` (_: Term.Param) `:parent:` (_: Template) =>
-        Some(top)
-      // val/def/var/type definitions or declarations
-      case (top: Type.With) `:parent:` (_: Defn | _: Decl) =>
+      // self types, params, val/def/var/type definitions or declarations
+      case (top: Type.With) `:parent:` (_: Defn | _: Decl | _: Term.Param) =>
```

Comment:
Just tried and it didn't come out great, need tree printer to properly fix this

Reply:
Yeah I tried removing the `parent` guard when playing around with my original PR and it ended up wrapping after the second `with` sometimes. I was thinking you need to at least make sure that `parent` is not another `Type.With` but `TreeOps.topTypeWith` should guarantee that 🤔

Reply:
Life is a lot easier in https://github.com/scalacenter/scala-syntax where we print code by traversing the tree instead of visiting the parent chain up from tokens as we do here.